module/os/linux/zfs/spa_misc_os.c

```diff
@@ -42,6 +42,17 @@
 #include <sys/kstat.h>
 #include "zfs_prop.h"
 
+
+int
+param_set_deadman_failmode(const char *val, zfs_kernel_param_t *kp)
+{
+	int error;
+
+	error = param_set_deadman_failmode_common(val);
+
+	return (error || param_set_charp(val, kp));
```

Comment:
This isn't quite right, we need the return value from `param_set_charp()` returned.  Let's keep it simple:

```
        int error = -param_set_deadman_failmode_common(val);
        if (error == 0)
                error = param_set_charp(val, kp);

        return (error);
```